src/tcp.h

```diff
@@ -35,21 +42,32 @@ struct enftun_tcp
         struct sockaddr remote_addr;
         char _remote_addr_pad[MAX_SOCKADDR_LEN];
     };
+
+    struct enftun_tcp_ops ops;
 };
 
+struct enftun_tcp_native
+{
+    struct enftun_tcp socket;
```

Comment:
Rename the `struct enftun_tcp` member to `base` instead of `socket`.  It's common way to indicate that it is the "base class" of `enftun_tcp_native`.  It also avoids weird code like `sock.socket.fd`, which becomes the more readable `sock.base.fd`.